### DISCUSSION TITLE
Can't get Syntax Highlighting to work for Ruby in Emacs

### DISCUSSION THREAD

**Discussion by goblynwtf** [emacs-tree-sitter/elisp-tree-sitter#160] [Q&A]
When I open a ruby file, I can see that Tree-Sitter-Hl mode is active, but none of the syntax highlighting has changed. But if I manually turn off it via M-X and then turn on again it start works. What am I doing wrong? How to force syntax highlighting automatically?

**Comment by elzorrorebelde** [comment]
can you share your configuration, please?

**Comment by goblynwtf** [comment]
https://github.com/0xk175un3/emacs.d here it is

**Comment by elzorrorebelde** [comment]
> https://github.com/0xk175un3/emacs.d here it is

Please note that `tree-sitter-hl-mode`:

> Enabling this automatically enables tree-sitter-mode in the buffer.
> 
> To enable this automatically whenever tree-sitter-mode is enabled:
> 
>  ```
>  (add-hook 'tree-sitter-after-on-hook #'tree-sitter-hl-mode)
>  ```

so you should modify the entire code block of your tree-sitter configuration to:

```
(use-package tree-sitter
  :ensure t
  :config
  (add-hook 'tree-sitter-after-on-hook #'tree-sitter-hl-mode)
  (add-hook 'ruby-mode-hook #'tree-sitter-mode))
```

I use Doom Emacs as my basis for my configuration and I have emacs-tree-sitter [configure like this](https://github.com/shackra/emacs/blob/master/modules/ui/tree-sitter/config.el). 

Please make changes and test if things work.

**Comment by goblynwtf** [comment]
Tried both variants, but sadly it doesn't work.

**Comment by elzorrorebelde** [comment]
> Tried both variants, but sadly it doesn't work.

keep the configuration with those corrections, further investigation on the rest of it may be necessary
